src/client/api/types.ts

```diff
@@ -94,6 +102,10 @@ export type PythonApi = {
      * @param resource file that determines which connection to return
      */
     getLanguageServer(resource?: InterpreterUri): Promise<ILanguageServer | undefined>;
+    /**
+     * Registers a visibility filter for the interpreter status bar.
+     */
+    registerInterpreterStatusFilter(filter: IInterpreterStatusbarVisibilityFilter): void;
```

Comment:
Hmm, if we make this API public, then its possible for any extension to hide this statusbar item. 
Most if not all of the API exposed by the Python extension is idempotent, however this one is not, it could be abused, not that any one will, however considering how VS Code works, they generally disallow such stuff. I.e. we won't want another extension hiding UI elements that it didn't contribute, however here' we're allowing that.

Just a thought.
I wouldn't ever want this to be made public, and considering that I'd prefer to just use VSC API in Python extension to hide the status bar when a notebook is active (this doesn't require any jupyter dependency).

Reply:
I thought the 'jupyter' api is private? And if our extension can hide the python extension's status bar, can't another extension then do the same thing?